server/lib/stripe.js

```diff
@@ -11,6 +11,12 @@ stripe.setMaxNetworkRetries(2);
 
 export default stripe;
 
+export const StripeCustomToken = token => {
+  const stripe = Stripe(token);
+  stripe.setMaxNetworkRetries(2);
+  return stripe;
+};
+
```

Comment:
I don't understand why we need this?

Reply:
Note that, 

https://github.com/opencollective/opencollective-api/blob/c87250acf2ed5a082565c44e3f9a24dce662eda9/cron/hourly/reconcile-transactions.js#L87

We create a new `Stripe` object here regardless of whether we use `config.stripe.secret` or `connectedAccount.token`. This creates a problem for the tests as I don't think if we use this approach it will work with things like, `sandbox.stub(stripe.issuing.transactions, 'list')...` etc. Since there's no way to attach to the new `Stripe` object that's been used in the `reconcile-transactions.js`. Therefore I made a small change there so as to reuse the `Stripe` object created within `stripe.js` when the `host.slug === 'opencollective'`. This way in our tests we could use `slug='opencollective'`,

 https://github.com/opencollective/opencollective-api/blob/6cc538908e2d5f654757ceaccac5d4c251b99407/test/cron/hourly/reconcile-transactions.test.js#L128

and the tests would simply be working fine. 

Also I felt that the creation of `Stripe` objects should be moved to the `stripe` lib.